semgrep-core/src/analyzing/Typing.ml

```diff
@@ -33,4 +33,14 @@ let get_resolved_type lang (vinit, vtype) =
       | Some { e = L (Null tok); _ } -> make_type "null" tok
       | Some { e = L (Imag (_, tok)); _ } -> make_type "imag" tok
       | Some { e = N (Id (_, { id_type; _ })); _ } -> !id_type
+      | Some
+          {
+            e = Call ({ e = IdSpecial (New, _); _ }, (_, ArgType tp :: _, _));
+            _;
+          } -> (
+          match lang with
```

Comment:
Yeah I wouldn't do the language check. This should e.g. work for Java too. If it matches that pattern I think it's safe toreturn `tp` as the type.